docs/creating-nodes/testing.md

```diff
@@ -0,0 +1,271 @@
+---
+layout: docs
+toc: creating-nodes-toc.html
+title: Testing
+---
+
+This guideline explains how to write and execute test code for node.
+
+Because nodes run in Node-RED, you cannot execute the processing of individual units of HTML or JavaScript to test processing.
+In addition, if you use manual testing, you need to stop and restart Node-RED after each test so that cleanup processing takes place.
+This makes testing quite time-consuming.
+Therefore, a tool called [Node Test Helper](https://github.com/node-red/node-red-node-test-helper) (hereafter "helper") has been developed for Node-RED.
+This tool provides a framework for node testing.
+By using this framework, you can easily run nodes at the code level.
+
+Node-RED uses the following libraries for testing, and we recommend their use when using the test code.
+You will need to have an understanding of the specifications of these libraries when creating test code.
+Read the documentation of each library and learn how to use it.
+
+| No. | Library | Purpose | URL |
+| --- | --- | ---| --- |
+| 1 | Mocha | Test framework | [https://mochajs.org](https://mochajs.org/) |
+| 2 | Should.js | Assertion | [https://shouldjs.github.io](https://shouldjs.github.io/) |
+| 3 | Sinon.js | Test double | [http://sinonjs.org](http://sinonjs.org/) |
```

Comment:
Currently we do use both sinon and nock - For simplicity and clarity should we just concentrate on one ?

Reply:
They serve different purposes. For general testing we use `sinon.js`. `Nock` is used specifically to mock up HTTP endpoints to test against. It may be helpful to make that distinction in this table.

Reply:
I added the notes in the commit, https://github.com/node-red/node-red.github.io/pull/109/commits/80fefaa9aac9eca4ab92af1b45c479ba68f1d060.